src/main/java/ch/admin/bag/covidcertificate/gateway/client/internal/DefaultIdentityAuthorizationClient.java

```diff
@@ -66,16 +67,23 @@ private boolean checkUserExists(QueryUsersResponse response) {
         }
     }
 
-    private boolean checkUserRoleExists(QueryUsersResponse response) {
+    protected boolean hasUserRoleSuperUserOrCreator(QueryUsersResponse response) {
         try {
-            List<Authorization> authorizations = response.getReturns().get(0)
-                    .getProfiles().get(0)
-                    .getAuthorizations();
-            return (authorizations.stream().anyMatch(authorization ->
-                    authorization.getRole().getExtId().equals(ROLE) || authorization.getRole().getExtId().equals(ROLE_SUPERUSER)));
+            AtomicBoolean result = new AtomicBoolean(false);
+            response.getReturns().forEach(
```

Comment:
I would recommend you to try with the snippet below, if it's even correct, for the following reasons:

- **filter and forEach**: iterate trough the entire list to apply the predicate or process the element

- **anyMatch**: it may not evaluate all elements of the stream, it stops searching when the result is definitive

```
return response.getReturns().stream().anyMatch(
                    user -> user.getProfiles().stream().anyMatch(
                            profile -> profile.getState().equals(ProfileState.ACTIVE) && profile.getAuthorizations().stream().anyMatch(this::isRoleSuperUserOrCreator)));
```

PLUS:
```
 ...profile -> result.set(profile.getAuthorizations().stream().anyMatch(
                            this::isRoleSuperUserOrCreator))...
```
I suspect this could return a wrong result as the previous loop (forEach) of the profile list continue to iterate and may override a match result (true) with the latest element evaluation result which could be a false.

Reply:
Hi, thanks for the review and hint. I changed it accordingly and added two tests that show, that the implementation works even if two profiles are active.